embedded_sass.proto

```diff
@@ -4,155 +4,176 @@
-    INDENTED = 1;
-
-    // Plain CSS syntax that doesn't support any special Sass features.
-    CSS = 2;
+// The wrapper type for all messages sent from the host to the compiler. This
+// provides a `oneof` that makes it possible to determine the type of each
+// inbound message.
+message InboundMessage {
+  // A request that compiles an entrypoint to CSS.
+  message CompileRequest {
+    // This compilation's request id. This is included in messages sent from the
+    // compiler to the host.
+    uint32 id = 1;
+
+    // Possible syntaxes for a Sass stylesheet.
+    enum Syntax {
+      // The CSS-superset `.scss` syntax.
+      SCSS = 0;
+
+      // The indented `.sass` syntax.
+      INDENTED = 1;
+
+      // Plain CSS syntax that doesn't support any special Sass features.
+      CSS = 2;
+    }
+
+    // An input stylesheet provided as plain text, rather than loaded from the
+    // filesystem.
+    message StringInput {
+      // The contents of the stylesheet.
+      string source = 1;
+
+      // The location from which `source` was lodaed. If this is empty, it
+      // indicates that the URL is unknown.
+      string url = 2;
+
+      // The syntax to use to parse `source`.
+      Syntax syntax = 3;
+    }
+
+    // The input stylesheet to parse. Mandatory.
+    oneof input {
+      // A stylesheet loaded from its contents.
+      StringInput string = 2;
+
+      // A stylesheet loaded from the given path on the filesystem.
+      string path = 3;
+    }
+
+    // Possible ways to format the CSS output. The compiler is not required to
+    // support all possible options; if the host requests an unsupported style,
+    // the compiler should choose the closest supported style.
+    enum OutputStyle {
+      // Each selector and declaration is written on its own line.
+      EXPANDED = 0;
+
+      // The entire stylesheet is written on a single line, with as few
+      // characters as possible.
+      COMPRESSED = 1;
+
+      // CSS rules and declarations are indented to match the nesting of the
+      // Sass source.
+      NESTED = 2;
+
+      // Each CSS rule is written on its own single line, along with all its
+      // declarations.
+      COMPACT = 3;
+    }
+
+    // How to format the CSS output.
+    OutputStyle style = 4;
+
+    // Whether to generate a source map. Note that this will *not* add a source
+    // map comment to the stylesheet; that's up to the host or its users.
+    bool source_map = 5;
   }
 
-  // An input stylesheet provided as plain text, rather than loaded from the
-  // filesystem.
-  message StringInput {
-    // The contents of the stylesheet.
-    string source = 1;
-
-    // The location from which `source` was lodaed. If this is empty, it
-    // indicates that the URL is unknown.
-    string url = 2;
-
-    // The syntax to use to parse `source`.
-    Syntax syntax = 3;
+  // The wrapped message. Mandatory.
+  oneof message {
+    CompileRequest compileRequest = 1;
```

Comment:
oneof with a single choice? Am I missing something here?
or it's just because we'll need it anyway since we're going to have more than one inbound message type (which is entirely reasonable)?

Reply:
> or it's just because we'll need it anyway since we're going to have more than one inbound message type (which is entirely reasonable)?

Yeah, that's exactly right. Eventually we'll have multiple inbound messages (such as responses to function and importer requests).